src/Service/Modules/Reports/PartsReceivedReportModule.cs

```diff
@@ -55,5 +58,22 @@ private async Task GetReport(HttpRequest req, HttpResponse res)
 
             await res.Negotiate(results);
         }
+
+        private async Task GetExport(HttpRequest req, HttpResponse res)
+        {
+            var options = new PartsReceivedReportRequestResource
+                {
+                    Supplier = req.Query.As<int?>("supplier"),
+                    FromDate = req.Query.As<string>("fromDate"),
+                    ToDate = req.Query.As<string>("toDate"),
+                    Jobref = req.Query.As<string>("jobref"),
+                    OrderBy = req.Query.As<string>("orderBy"),
+                    IncludeNegativeValues = req.Query.As<bool>("includeNegativeValues")
+                };
+            
+            var csv = this.reportFacadeService.GetReportCsv(options);
+
+            await res.FromCsv(csv, "parts_received.csv");
```

Comment:
all of the csv -> stream business is just replaced with this .FromCsv(..) extension to the HttpResponse